pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -335,7 +335,7 @@ private void processNewSegment(String tableNameWithType, SegmentMetadata segment
     }
 
     try {
-      _pinotHelixResourceManager.assignTableSegment(tableNameWithType, segmentMetadata.getName());
+      _pinotHelixResourceManager.assignTableSegment(tableNameWithType, segmentMetadata.getName(), _controllerMetrics);
```

Comment:
Very surprised to see we don't have access to controller metrics in the `PinotHelixResourceManager`. Can you pass it in when calling `PinotHelixResourceManager.start()`?

Reply:
looks like _pinotHelixResourceManager is initialized (as part of init() method), before _controllerMetrics (as part of start() method), I'll see if that can be reordered.